app/components/user-profile-permissions.js

```diff
@@ -1,197 +1,187 @@
-    }
-    let currentYear = parseInt(moment().format('YYYY'), 10);
-    const currentMonth = parseInt(moment().format('M'), 10);
+    let currentYear = Number(moment().format('YYYY'));
+    const currentMonth = Number(moment().format('M'));
     if(currentMonth < 6){
       currentYear--;
     }
-    let defaultYear = years.find(year => parseInt(year.get('id'), 10) === currentYear);
-    if(isEmpty(defaultYear)){
-      defaultYear = years.get('lastObject');
+    let selectedYear = this.academicYears.find(year => Number(year.id) === currentYear);
+    if(!selectedYear){
+      selectedYear = this.academicYears.get('lastObject');
     }
-
-    return defaultYear;
-  }),
-
-  isDirectingSchool: computed('user.directedSchools.[]', 'selectedSchool', async function () {
-    const selectedSchool = await this.selectedSchool;
-    const user = this.user;
-    const ids = user.hasMany('directedSchools').ids();
-    return ids.includes(selectedSchool.get('id'));
-  }),
-
-  isAdministeringSchool: computed('user.administeredSchools.[]', 'selectedSchool', async function () {
-    const selectedSchool = await this.selectedSchool;
-    const user = this.user;
-    const ids = user.hasMany('administeredSchools').ids();
-    return ids.includes(selectedSchool.get('id'));
-  }),
-
-  directedPrograms: computed('user.directedPrograms.[]', 'selectedSchool', async function () {
-    const selectedSchool = await this.selectedSchool;
-    const user = this.user;
-    const directedPrograms = await user.get('directedPrograms');
-
-    return filter(directedPrograms.toArray(), async program => {
-      const school = await program.get('school');
+    this.selectedYearId = selectedYear?.id;
+    if (map.defaultSchool) {
+      yield this.changeSchool.perform(map.defaultSchool.id);
+    } else if (this.schools.length) {
+      yield this.changeSchool.perform(this.schools.sortBy('title')[0].id);
+    }
+  }
+
+  @restartableTask
+  *changeSchool(schoolId) {
+    this.selectedSchool = this.schools.findBy('id', schoolId);
+    this.isDirectingSchool = this.args.user.hasMany('directedSchools').ids().includes(schoolId);
+    this.isAdministeringSchool = this.args.user.hasMany('administeredSchools').ids().includes(schoolId);
+
+    const map = yield hash({
+      directedPrograms: this.getDirectedPrograms(this.selectedSchool),
+      directedProgramYears: this.getDirectedProgramYears(this.selectedSchool),
+      directedCourses: this.getDirectedCourses(this.selectedSchool),
+      administeredCourses: this.getAdministeredCourses(this.selectedSchool),
+      instructedCourses: this.getInstructedCourses(this.selectedSchool),
+      studentAdvisedCourses: this.getStudentAdvisedCourses(this.selectedSchool),
+      administeredSessions: this.getAdministeredSessions(this.selectedSchool),
+      instructedSessions: this.getInstructedSessions(this.selectedSchool),
+      studentAdvisedSessions: this.getInstructedSessions(this.selectedSchool),
```

Comment:
should be something like `this.getStudenAdvisedSessions`

Reply:
Thanks! Fixed.